gdal/ogr/ogrsf_frmts/geojson/ogrgeojsonwriter.cpp

```diff
@@ -1502,9 +1502,24 @@ char* OGR_G_ExportToJsonEx( OGRGeometryH hGeometry, char** papszOptions )
     oOptions.nCoordPrecision = nCoordPrecision;
     oOptions.nSignificantFigures = nSignificantFigures;
 
+    // If the CRS has latitude, longitude axis order, and the data axis to SRS
+    // axis mapping doesn't change that order, then swap X and Y values.
+    bool bHasSwappedXY = false;
+    const auto poSRS = poGeometry->getSpatialReference();
+    if( poSRS && poSRS->EPSGTreatsAsLatLong() &&
+        poSRS->GetDataAxisToSRSAxisMapping() == std::vector<int>{1, 2} )
+    {
+        poGeometry->swapXY();
+        bHasSwappedXY = true;
+    }
+
     json_object* poObj =
        OGRGeoJSONWriteGeometry( poGeometry, oOptions );
 
+    // Unswap back
+    if( bHasSwappedXY )
```

Comment:
Ok, I see now why this fix is unpleasant. Is this unswap here to avoid compatibility problems?

Reply:
It is an implementation detail to avoid that the caller of the function sees the geometry altered. The caller of the function just wants the geometry to be exported to GeoJSON, likely not modified.